language/evm/examples/sources/ERC721.move

```diff
@@ -0,0 +1,212 @@
+#[contract]
+/// An implementation of the ERC-721 Non-Fungible Token Standard.
+module Evm::ERC721 {
+    use Evm::Evm::{sender, self, sign, emit, isContract};
+    use Evm::Result;
+    use Evm::IERC721Receiver;
+    use Evm::IERC721;
+    use Evm::IERC165;
+    use Evm::Table::{Self, Table};
+    use Std::Errors;
+
+    #[event]
+    struct Transfer {
+        from: address,
+        to: address,
+        tokenId: u128,
+    }
+
+    #[event]
+    struct Approval {
+        owner: address,
+        approved: address,
+        tokenId: u128,
+    }
+
+    #[event]
+    struct ApprovalForAll {
+        owner: address,
+        operator: address,
+        approved: bool,
+    }
+
+    #[storage]
+    /// Represents the state of this contract. This is located at `borrow_global<State>(self())`.
+    struct State has key {
+        owners: Table<u128, address>,
```

Comment:
> Was wondering what's your take for using `borrow_global` instead of `Table` as much as possible to follow more traditional Move paradigm? IMHO this `balance` field should be represented as resource even at storage.

The problem is that we can key resources only by (static) type and address. Here the key is u256 (I think its currently u128 only because we have no full support for this type yet).

Reply:
@runtian-zhou, Wolfgang already answered it, but if I repeat, ... I think we need a data structure for general tables anyway (from an arbitrary type `K` to an arbitrary type `V`) Conceptually, Move offers a global "table" from `address` to `V` where `V` is an arbitrary resource type, and the table can be accessed by the global functions (e.g., `borrow_global`, `move_to`). However, the ERC examples seems to require more general tables (e.g., `u256->address`, and `u256->(address->u256)`). Although it's possible to implement them now in Move by storing key-value pairs in `vector` without using `Table`, it would be very inefficient and not comparable to the performance of the Solidity implementation. Moreover, Move requires `signer` for the `move_to` operation in order to add a new entry to the global table (i.e., global storage). It seems to be too restrictive in the context of Ethereum and EVM.